libraries/lib-utility/MemoryX.h

```diff
@@ -574,4 +574,16 @@ auto Visit(Visitor &&vis, Variant &&var)
       std::forward<Visitor>(vis), std::forward<Variant>(var) );
 }
 
+
+template <typename LType, typename RType>
+auto RoundUp(LType numerator, RType denominator) noexcept
+{
+   static_assert(std::is_integral_v<LType>);
+   static_assert(std::is_integral_v<RType>);
+
+   const auto result = numerator / denominator;
+
+   return result * denominator == numerator ? result : result + 1;
```

Comment:
https://en.cppreference.com/w/cpp/language/operator_arithmetic

If the denominator is zero, there is UB not an exception, so `noexcept` is strictly speaking correct.

If the values are not both positivie, what should happen?  (Division must truncate toward zero in C++11 and later; so if this case matters, then it is already "rounded up" and adding 1 is wrong.)

If those cases aren't supposed to happen, add a doxygen `@pre` that numerator is nonnegative and denominator is positive.  (Precondition means "it's callers responsibility to exclude those cases, not mine to handle them")

Ignoring the sign question, I would simplify:

`return (numerator + denominator - 1) / denominator;`

Reply:
Great catch, thank you!

Reply:
Your version is faster, but does not consider the possible integer overflow. I think I will just add Unsafe to the function name, though :-)